AirCasting/CoreData/MeasurementStreamStorage.swift

```diff
@@ -31,13 +33,25 @@ final class CoreDataMeasurementStreamStorage: MeasurementStreamStorage {
 
     private let persistenceController: PersistenceController
     private lazy var updateSessionParamsService = UpdateSessionParamsService()
-
+    private let context: NSManagedObjectContext
+    
     init(persistenceController: PersistenceController) {
         self.persistenceController = persistenceController
+        self.context = persistenceController.editContext()
     }
-
+    
+    func save() throws {
+        DispatchQueue.main.async {
```

Comment:
this cannot be on main queue. The context is created as `.editContext()`, so it's queue will probably not be a main one. Use a [perform(_:)](https://developer.apple.com/documentation/coredata/nsmanagedobjectcontext/1506578-perform) or [performAndWait(_:)](https://developer.apple.com/documentation/coredata/nsmanagedobjectcontext/1506364-performandwait) in order to dispatch properly